### PR TITLE
upgrade to fsdb v0.3.2

### PR DESCRIPTION
**Pull request by ael-code**

### CODE REVIEW

**Comment by boyska** [comment]
well, I don't think this is a great approach. I think that this is the right moment to polish the code and make it unit-testable. Right now, it's not. I have really no hurry in upgrading to fsdb v0.3.2, and prefer to wait a bit and have better code quality in libreant than just "patching" the new method names.

The point is that a real, meaningful migration to fsdb v0.3 also means using buffers instead of paths. Doing so is a not-so-small change, so proper testing is required. Look at my branch (I left a comment on #132 about it), I hope you like it.

**Comment by ael-code** [comment]
> well, I don't think this is a great approach

No. It's necessary. Fsdb had a big bug in the previous version.

>  I think that this is the right moment to polish the code and make it unit-testable. Right now, it's not. I have really no hurry in upgrading to fsdb v0.3.2, and prefer to wait a bit and have better code quality in libreant than just "patching" the new method names.

polish the code and make it unit-testable have nothing to do with this commit, really nothing.

> The point is that a real, meaningful migration to fsdb v0.3 also means using buffers instead of paths.

In some function that we use paths are better then streams: fsdb has some optimization for paths (buffer sizing) and flask [does not provide some functionality with streams](http://flask.pocoo.org/docs/0.10/api/#flask.send_file). In addition it's easy to mess up things with seekable streams.

**Comment by boyska** [comment]
On Tue, Mar 24, 2015 at 02:36:34PM -0700, ael wrote:
>> well, I don't think this is a great approach
>
>No. It's necessary. Fsdb had a big bug in the previous version.

which one? so maybe we should add the "bug" label to #132

>>  I think that this is the right moment to polish the code and make it unit-testable. Right now, it's not. I have really no hurry in upgrading to fsdb v0.3.2, and prefer to wait a bit and have better code quality in libreant than just "patching" the new method names.
>
>polish the code and make it unit-testable have nothing to do with this commit, really nothing.

indeed :) the point is that I don't want to lose such an occasion. If
you promise ( :P ) to be willing to work on those goals anyway, this
pull request seems fine to me.

>> The point is that a real, meaningful migration to fsdb v0.3 also means using buffers instead of paths.
>
>In some function that we use paths are better then streams: fsdb has some optimization for paths (buffer sizing) and flask [does not provide some functionality with streams](http://flask.pocoo.org/docs/0.10/api/#flask.send_file). In addition it's easy to mess up things with seekable streams.

You are talking about the X-Sendfile feature? is it important at all?
and what do you mean by "easy to mess up things" ? I don't think that
for example it is difficult to use buffers when offering book download.

**Comment by ael-code** [comment]
On 03/24/2015 10:49 PM, BoySka wrote:
> On Tue, Mar 24, 2015 at 02:36:34PM -0700, ael wrote:
>>> well, I don't think this is a great approach
>>
>> No. It's necessary. Fsdb had a big bug in the previous version.
> 
> which one? so maybe we should add the "bug" label to #132

https://github.com/ael-code/pyFsdb/commit/4469a513ea68ce2a1b5211ce18cc10bfb1584e73

> 
>>>  I think that this is the right moment to polish the code and make it unit-testable. Right now, it's not. I have really no hurry in upgrading to fsdb v0.3.2, and prefer to wait a bit and have better code quality in libreant than just "patching" the new method names.
>>
>> polish the code and make it unit-testable have nothing to do with this commit, really nothing.
> 
> indeed :) the point is that I don't want to lose such an occasion. If
> you promise ( :P ) to be willing to work on those goals anyway, this
> pull request seems fine to me.
> 

I promise.

>>> The point is that a real, meaningful migration to fsdb v0.3 also means using buffers instead of paths.
>>
>> In some function that we use paths are better then streams: fsdb has some optimization for paths (buffer sizing) and flask [does not provide some functionality with streams](http://flask.pocoo.org/docs/0.10/api/#flask.send_file). In addition it's easy to mess up things with seekable streams.
> 
> You are talking about the X-Sendfile feature? is it important at all?

I'm talking about X-Sendfile and mimetype guessing. They are not so
important, it was just an example.

**Comment by boyska** [comment]
>>> No. It's necessary. Fsdb had a big bug in the previous version.
>>
>> which one? so maybe we should add the "bug" label to #132
>
>https://github.com/ael-code/pyFsdb/commit/4469a513ea68ce2a1b5211ce18cc10bfb1584e73

ok, I add the 'bug' label. This changes everything!

By the way, I believe (but I am not sure) that a release that needs to
update the dependencies should not be a "patch" (as in
http://semver.org/ ), but at least a minor. So I propose that after your
commit get merged, we will bump to 0.2

>>>> The point is that a real, meaningful migration to fsdb v0.3 also means using buffers instead of paths.
>>>
>>> In some function that we use paths are better then streams: fsdb has some optimization for paths (buffer sizing)
>I'm talking about X-Sendfile and mimetype guessing. They are not so
>important, it was just an example.

mimetype guessing is slightly more important, IMHO, so ok.

summing it up: I will soon test your commit and, if it behaves
correctly, I will merge. After that I will rebase my refactoring and
submit it to your attention.

**Comment by ael-code** [comment]
> summing it up: I will soon test your commit and, if it behaves correctly, I will merge. After that I will rebase my refactoring and submit it to your attention.

waiting :)

**Comment by boyska** [comment]
>> summing it up: I will soon test your commit and, if it behaves correctly, I will merge.
>
>waiting :)

I'm sorry, I'm really busy in this period. If anyone else wants to step
ahead and test your commit, I'll appreciate this.
Otherwise, please continue to wait... and ping me from time to time!

File: webant/webant.py [hunk 1]
```diff
@@ -145,18 +146,18 @@ def upload():
...
-            tmpFile, tmpFilePath = tempfile.mkstemp()
+            tmpFileFd, tmpFilePath = tempfile.mkstemp()
             upFile.save(tmpFilePath)
             fileInfo = {}
             fileInfo['name'] = secure_filename(upFile.filename)
             fileInfo['size'] = os.path.getsize(tmpFilePath)
             fileInfo['mime'] = upFile.mimetype
             fileInfo['notes'] = request.form[upName+'_notes']
-            fileInfo['sha1'] = Fsdb.fileDigest(tmpFilePath, algorithm="sha1")
+            fileInfo['sha1'] = calc_file_digest(tmpFilePath, algorithm="sha1")
```
Comment (hellais) [inline webant/webant.py]: I would have called this simply `file_digest` and `digest`, but antani :P

File: webant/webant.py [hunk 2]
```diff
@@ -213,7 +214,7 @@ def download_book(bookid, fileid):
                 except:
                     app.logger.warn("Cannot increment download count",
                                     exc_info=1)
-                return send_file(app.fsdb.getFilePath(file['fsdb_id']),
+                return send_file(app.fsdb.get_file_path(file['fsdb_id']),
```
Comment (hellais) [inline webant/webant.py]: Same here having the get in front of it is soo java :P s/get_file_path/file_path/. The antani rule is valid also here.

**Comment by ael-code** [comment]
both your comments are correct but related to fsdb and not to libreant. 
Anyway did you test it? Can we merge it?

**Comment by boyska** [comment]
As discussed yesterday, assigned to @leophys 

Un applauso per l'accollo!

**Comment by hellais** [comment]
:clap: :clap: :tada:

**Comment by leophys** [comment]
Tested. It works.

**Pull request merged**